From 2bde25accb4aa584ef32292710cc8e1a63d62f1c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marco=20Trevisan=20=28Trevi=C3=B1o=29?= <mail@3v1n0.net>
Date: Mon, 16 Mar 2020 12:33:57 +0100
Subject: [PATCH 30/37] docking: Apply the height/width dash constraint after
 adding to main uiGroup

When calling Main.uiGroup.add_child(this); for the DockedDash we may end up
into an infinite loop caused by the previously set bind constraint (as per
the mutter upstream change 4f8e518d4).

So, just set the constraint once we've added the actor to the main uiGroup

Fixes #1117
---
 docking.js | 16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/docking.js b/docking.js
index ec4d19c..15defd0 100644
--- a/docking.js
+++ b/docking.js
@@ -251,13 +251,6 @@ var DockedDash = GObject.registerClass({
         });
         this._box.connect('notify::hover', this._hoverChanged.bind(this));
 
-        // Create and apply height constraint to the dash. It's controlled by this.height
-        this.constrainSize = new Clutter.BindConstraint({
-            source: this,
-            coordinate: this._isHorizontal?Clutter.BindCoordinate.WIDTH:Clutter.BindCoordinate.HEIGHT
-        });
-        this.dash.add_constraint(this.constrainSize);
-
         this._signalsHandler.add([
             // update when workarea changes, for instance if  other extensions modify the struts
             //(like moving th panel at the bottom)
@@ -378,6 +371,15 @@ var DockedDash = GObject.registerClass({
         else
             Main.layoutManager._trackActor(this._slider);
 
+        // Create and apply height/width constraint to the dash.
+        // It's controlled by this.height or this.width
+        this._constrainSize = new Clutter.BindConstraint({
+            source: this,
+            coordinate: this._isHorizontal ?
+                Clutter.BindCoordinate.WIDTH : Clutter.BindCoordinate.HEIGHT
+        });
+        this.dash.add_constraint(this._constrainSize);
+
         // Set initial position
         this._resetPosition();
 
-- 
2.26.0

